src/acl.c

```diff
@@ -2602,7 +2602,8 @@ void aclCommand(client *c) {
     } else if (!strcasecmp(sub,"getuser") && c->argc == 3) {
         user *u = ACLGetUserByName(c->argv[2]->ptr,sdslen(c->argv[2]->ptr));
         if (u == NULL) {
-            addReplyNull(c);
+            addReplyErrorFormat(c,
+                    "Can not find user '%s' in ACL",(char*)c->argv[2]->ptr);
```

Comment:
i see this null response is not documented, but it was always like that (not a recent change).
the change makes sense (to return an error), but i'm not certain we wanna change it.

Reply:
Now, if the client sends command: ACL getuser testuser, which is not in ACL, the client gets the following result:

![image](https://user-images.githubusercontent.com/51993843/151251038-34a74c81-50f5-4d4b-a782-e7cf9337d7ea.png)

The updated error message could give client a more reasonable feeback as below:

![image](https://user-images.githubusercontent.com/51993843/151251501-ed023b5f-72f6-4f9c-87bc-1054c1111ec6.png)

Reply:
ERR is more nice (+1 for ERR), but it's not backward compatible (+1 for nil). OTOH, the nil reply is undocumented (-1 for nil). I don't think anyone (or any admin tool) relies on this nil reply, but it's not impossible.

Most commands return nil or empty array on non-existing entities, e.g. `GET nokey`, `COMMAND INFO xyz`, `CONFIG GET xyz`. The nil reply of ACL GETUSER is actually consistent with these. (+1 for nil)

Total score: ERR=1, nil=2. :grin:

Reply:
I think this is a backwards breaking change. If someone wrote any automation like, `if not getUser(foo): setUser(foo, whatever)`, this will go from working to throwing an error. I feel strongly we have to keep it as is. 

P.S. I don't think that math adds up. It looks like ERR=1 and nil=1 :P

Reply:
we discussed this in a core-team meeting, we concluded that returning NIL in this case isn't really wrong.
it's similar to returning 0 on EXISTS, or NIL on SPOP on a missing key.
i.e. GETUSER can be used to check if a user exists.
in anyway, we don't like to introduce a breaking change about it, and instead rather document the existing behavior.
@hwware can you update this PR, and create a redis-doc PR?

Reply:
> we discussed this in a core-team meeting, we concluded that returning NIL in this case isn't really wrong. it's similar to returning 0 on EXISTS, or NIL on SPOP on a missing key. i.e. GETUSER can be used to check if a user exists. in anyway, we don't like to introduce a breaking change about it, and instead rather document the existing behavior. @hwware can you update this PR, and create a redis-doc PR?

Sure, i will update this PR, thanks

Reply:
> P.S. I don't think that math adds up. It looks like ERR=1 and nil=1 :P

(@madolson Check again. :-) The last +1 for nil is in the second paragraph, the one about other commands returning nil for object-not-found. Either way, my conclusion was the same as yours, though my method was more pseudo-scientific. :P)